src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
```

Comment:
The field private static final int def_capacity = 10; can be renamed to private static final int DEFAULT_CAPACITY = 10; for better readability.

The empty_arraydata array should be declared as private static final Object[] EMPTY_ARRAY_DATA = {}; to follow the standard naming conventions.